### ISSUE TITLE
BUSINESS_RULE violations in the LPR3 administration client

### ISSUE THREAD

**Issue by RasmusThernoe** [scandihealth/lpr3-docs#94]
I create an episode of care using the LPR3 testservice - with a known business rule violation.

I get this response:
```xml
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope">
   <soap:Header>
      <ns10:Header xmlns="urn:oasis:names:tc:ebxml-regrep:xsd:rim:3.0" xmlns:ns10="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd" xmlns:ns11="http://www.nsi.dk/hsuid/2016/08/hsuid-1.1.xsd" xmlns:ns2="urn:oasis:names:tc:ebxml-regrep:xsd:rs:3.0" xmlns:ns3="urn:oasis:names:tc:ebxml-regrep:xsd:lcm:3.0" xmlns:ns4="urn:ihe:iti:xds-b:2007" xmlns:ns5="urn:oasis:names:tc:ebxml-regrep:xsd:query:3.0" xmlns:ns6="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" xmlns:ns7="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:ns8="http://www.w3.org/2000/09/xmldsig#" xmlns:ns9="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd">
         <ns10:SecurityLevel>3</ns10:SecurityLevel>
         <ns10:Linking>
            <ns10:MessageID>f1c8fe24c0a8380100c79bd579cb058a</ns10:MessageID>
         </ns10:Linking>
         <ns10:FlowStatus>flow_finalized_succesfully</ns10:FlowStatus>
      </ns10:Header>
   </soap:Header>
   <soap:Body>
      <ns2:RegistryResponse status="urn:oasis:names:tc:ebxml-regrep:ResponseStatusType:Success" xmlns="urn:oasis:names:tc:ebxml-regrep:xsd:rim:3.0" xmlns:ns10="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd" xmlns:ns11="http://www.nsi.dk/hsuid/2016/08/hsuid-1.1.xsd" xmlns:ns2="urn:oasis:names:tc:ebxml-regrep:xsd:rs:3.0" xmlns:ns3="urn:oasis:names:tc:ebxml-regrep:xsd:lcm:3.0" xmlns:ns4="urn:ihe:iti:xds-b:2007" xmlns:ns5="urn:oasis:names:tc:ebxml-regrep:xsd:query:3.0" xmlns:ns6="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" xmlns:ns7="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:ns8="http://www.w3.org/2000/09/xmldsig#" xmlns:ns9="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd">
         <ns2:RegistryErrorList highestSeverity="urn:oasis:names:tc:ebxml-regrep:ErrorSeverityType:Warning">
            <ns2:RegistryError codeContext="BUSINESS_RULE|||Henvisningstidspunkt for en henvisning, der knytter sig til et forløbselement, skal ligge før eller samtidig med forløbselementets starttidspunkt" errorCode="InvalidDocumentContent" location="830C4B22-C92C-5588-A9BA-41E93E626D15^087b70e5-c9a4-53fb-b67e-3298a9a96498|||//*[child::*[local-name()='id' and @root='830C4B22-C92C-5588-A9BA-41E93E626D15' and @extension='98259e90-86b6-5a3b-bc72-a721c4401ed0']]|||sds:lpr:3.0.4|||06.01" severity="urn:oasis:names:tc:ebxml-regrep:ErrorSeverityType:Warning"/>
         </ns2:RegistryErrorList>
      </ns2:RegistryResponse>
   </soap:Body>
</soap:Envelope>
```

An hour later I perform a search in the LPR3 administration client on "Vis Indberetninger". The episode of care has "Fejlttype= OK":
![fejltype](https://user-images.githubusercontent.com/36242392/39123718-fea59ccc-46f8-11e8-8feb-7e7346ca74ef.png)

I also perform a search in the LPR3 administration client on "Vis fejl". No search results found:
![fejlliste](https://user-images.githubusercontent.com/36242392/39123731-05050922-46f9-11e8-8274-b0f7cf17e12f.png)

According to the release note of UL4: "The LPR3 Accumulated Error List will be generated **every 5 minutes** for testing purposes".

**Comment by RasmusThernoe** [comment]
I have now created a new episode of care at 10.42 a.m. which also has the business rule:

```xml
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope">
   <soap:Header>
      <ns10:Header xmlns="urn:oasis:names:tc:ebxml-regrep:xsd:rim:3.0" xmlns:ns10="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd" xmlns:ns11="http://www.nsi.dk/hsuid/2016/08/hsuid-1.1.xsd" xmlns:ns2="urn:oasis:names:tc:ebxml-regrep:xsd:rs:3.0" xmlns:ns3="urn:oasis:names:tc:ebxml-regrep:xsd:lcm:3.0" xmlns:ns4="urn:ihe:iti:xds-b:2007" xmlns:ns5="urn:oasis:names:tc:ebxml-regrep:xsd:query:3.0" xmlns:ns6="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" xmlns:ns7="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:ns8="http://www.w3.org/2000/09/xmldsig#" xmlns:ns9="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd">
         <ns10:SecurityLevel>3</ns10:SecurityLevel>
         <ns10:Linking>
            <ns10:MessageID>00e45abac0a8380101af224d346b0b2c</ns10:MessageID>
         </ns10:Linking>
         <ns10:FlowStatus>flow_finalized_succesfully</ns10:FlowStatus>
      </ns10:Header>
   </soap:Header>
   <soap:Body>
      <ns2:RegistryResponse status="urn:oasis:names:tc:ebxml-regrep:ResponseStatusType:Success" xmlns="urn:oasis:names:tc:ebxml-regrep:xsd:rim:3.0" xmlns:ns10="http://www.medcom.dk/dgws/2006/04/dgws-1.0.xsd" xmlns:ns11="http://www.nsi.dk/hsuid/2016/08/hsuid-1.1.xsd" xmlns:ns2="urn:oasis:names:tc:ebxml-regrep:xsd:rs:3.0" xmlns:ns3="urn:oasis:names:tc:ebxml-regrep:xsd:lcm:3.0" xmlns:ns4="urn:ihe:iti:xds-b:2007" xmlns:ns5="urn:oasis:names:tc:ebxml-regrep:xsd:query:3.0" xmlns:ns6="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" xmlns:ns7="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:ns8="http://www.w3.org/2000/09/xmldsig#" xmlns:ns9="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd">
         <ns2:RegistryErrorList highestSeverity="urn:oasis:names:tc:ebxml-regrep:ErrorSeverityType:Warning">
            <ns2:RegistryError codeContext="BUSINESS_RULE|||Henvisningstidspunkt for en henvisning, der knytter sig til et forløbselement, skal ligge før eller samtidig med forløbselementets starttidspunkt" errorCode="InvalidDocumentContent" location="A674574F-0F55-54F1-8CC9-685FDB83259F^51152d4d-df7b-5ee1-9c53-bca5f035e2a5|||//*[child::*[local-name()='id' and @root='A674574F-0F55-54F1-8CC9-685FDB83259F' and @extension='d5f23087-1af8-5972-9fda-6190638e3f66']]|||sds:lpr:3.0.4|||06.01" severity="urn:oasis:names:tc:ebxml-regrep:ErrorSeverityType:Warning"/>
         </ns2:RegistryErrorList>
      </ns2:RegistryResponse>
   </soap:Body>
</soap:Envelope>
``` 

This business rule violation is not displayed in the LPR3 administration client and furthermore not present in the folder 'lpr3-fejlliste' at ftp:
![image](https://user-images.githubusercontent.com/36242392/39296123-63b7fe0e-4940-11e8-9f0a-c947a3851cea.png)

Where is my business rule violation?

**Comment by RasmusThernoe** [comment]
Input files for creating second Episode of Care:

Header:  [header.zip](https://github.com/scandihealth/lpr3-docs/files/1955156/header.zip)

CDA document: [TC1-EpisodeOfCare-Create-Test.zip](https://github.com/scandihealth/lpr3-docs/files/1955158/TC1-EpisodeOfCare-Create-Test.zip)

**Comment by RasmusThernoe** [comment]
This is issue is blocking - we are testing our parser of the accumulated error list.

**Comment by TueCN** [comment]
Sorry for the late response.

It turns out this issue is caused by a bug that has also been found internally by DXC. The bug has been fixed with the latest deploy: [Release 3.1.0.674](https://github.com/scandihealth/lpr3-docs/milestone/11) 
> LPR-1671: Validation errors are not attached to a reporting element

I have retested your issue by sending a new report using the provided XML in TC1-EpisodeOfCare-Create-Test.zip, where setId and episode of care extension values have been changed. Now the validation errors are visible from the GUI, as expected.

Unfortunately the fix does not apply to existing submissions. Only new submissions made to the service have their validation errors correctly "attached".

You can see my new test submission by going to "Vis indberetninger" and search for forløbsid `2b9c38b3-b4cf-428d-9711-30c855ff171c`. 
The found episode of care matches the one in your XML and has validation error `601 | 06.01 sds:lpr:3.0.4 | Henvisningstidspunkt for en henvisning, der knytter sig til et forløbselement, skal ligge før eller samtidig med forløbselementets starttidspunkt`

**Comment by RasmusThernoe** [comment]
Thank you for the fix - it works.
